### PR TITLE
More accurate time logging for ImageEncoder and fix concurrent image processing corruption

### PR DESCRIPTION
**Pull request by irexyc**
## Motivation

https://github.com/InternLM/lmdeploy/issues/1759

### CODE REVIEW

File: lmdeploy/vl/engine.py [hunk 1]
```diff
@@ -104,23 +98,22 @@ def forward(self, inputs: List[Image]):
         """Model forward."""
         time_start = time.perf_counter()
         outputs = self.model.forward(inputs)
+        if isinstance(outputs[0], torch.Tensor):
```
Comment (RunningLeon) [inline lmdeploy/vl/engine.py:101]: do we still need to convert to cpu ?

**Review by grimoire — APPROVED** [review]

File: lmdeploy/vl/engine.py (hunk 1 above)
Comment (irexyc) [reply lmdeploy/vl/engine.py:101]: The modification here is mainly to solve the https://github.com/InternLM/lmdeploy/issues/1759

Comment (lvhan028) [reply lmdeploy/vl/engine.py:101]: Will it slow down the inference performance?

Comment (irexyc) [reply lmdeploy/vl/engine.py:101]: test with llava-v1.5-7b and profile_restful_api.py (PR 1662)

Move tensor to cpu doesn't affect the performance. But using asyncio.Task may affects the performance.

```
# with thread
number of prompt tokens: 248339
number of completion tokens: 240582
token throughput (completion token): 825.260 token/s
token throughput (prompt + completion token): 1677.128 token/s
RPS (request per second): 3.430 req/s
RPM (request per minute): 205.816 req/min

# with asyncio.Task
number of prompt tokens: 248339
number of completion tokens: 240582
token throughput (completion token): 808.719 token/s
token throughput (prompt + completion token): 1643.513 token/s
RPS (request per second): 3.362 req/s
RPM (request per minute): 201.691 req/min
```

Comment (grimoire) [reply lmdeploy/vl/engine.py:101]: If gpu utility is high, you can add
```
        await asyncio.get_event_loop().run_in_executor(None,
                                                       self.stream.synchronize)
```
after forward so other coroutine can use cpu during forward.

Comment (irexyc) [reply lmdeploy/vl/engine.py:102]: According to my test, wrapping forward is better to synchronize the stream. 

```python
outputs = await asyncio.get_event_loop().run_in_executor(
                None, self.forward, inputs)
# token throughput (completion token): 826.370 token/s
```

```python
outputs = self.model.forward(inputs)
if isinstance(outputs[0], torch.Tensor):
    stream = torch.cuda.current_stream(outputs[0].device)
    await asyncio.get_event_loop().run_in_executor(
                None, stream.synchronize)
# token throughput (completion token): 815.130 token/s
```

**Review by RunningLeon — APPROVED** [review]
LGTM

**Comment by lvhan028** [comment]
@irexyc Does this PR ONLY target #1759?

**Comment by irexyc** [comment]
>Does this PR ONLY target https://github.com/InternLM/lmdeploy/issues/1759?

and https://github.com/InternLM/lmdeploy/issues/1730

**Pull request merged**